transforms/StickbreakingLogistic/StickbreakingLogistic_functions.stan

```diff
@@ -0,0 +1,31 @@
+vector stickbreaking_logistic_simplex_constrain_lp(vector y) {
+  int N = rows(y) + 1;
+  vector[N] x;
+  real log_zi, log_xi;
+  real log_cum_prod = 0;
+  for (i in 1 : N - 1) {
+    log_zi = log_inv_logit(y[i] - log(N - i)); // logistic_lcdf(y[i] | log(N - i), 1)
+    log_xi = log_cum_prod + log_zi;
+    x[i] = exp(log_xi);
+    log_cum_prod += log1m_exp(log_zi);
+    target += log_xi;
+  }
+  x[N] = exp(log_cum_prod);
+  target += log_cum_prod;
+  return x;
+}
+
+vector stickbreaking_logistic_log_simplex_constrain_lp(vector y) {
+  int N = rows(y) + 1;
+  vector[N] log_x;
+  real log_z;
+  real log_cum_prod = 0;
+  for (i in 1 : (N - 1)) {
+    log_z = log_inv_logit(y[i] - log(N - i)); // logistic_lcdf(y[i] | log(N - i), 1)
```

Comment:
The built-in forms are usually more efficient as they produce more compact expression graphs for autodiff, but again ,no big deal here if you think this is clearer.

There's also a `cumulative_sum()` built-in, which could be used here.   It's also more efficient than the loop for the same reason---smaller autodiff graph and fewer virtual function calls in the backward pass.  Again, no big deal if you like it this way better.

Reply:
Ah, this will hopefully clear up some confusion I have about whether it's better in Stan to vectorize even if it means more array allocations. e.g. here `logistic_lcdf` is not vectorized, so to use that we would definitely define `log_z` inside the loop. But we could also eliminate the loop entirely and do this

```
  // vectorized logistic_lcdf(y[i] | log(N - i), 1)
  vector[N - 1] log_z = log_inv_logit(y - log(reverse(linspaced_vector(N - 1, 1, N - 1))));
  vector[N] log_x = append_row(log_z, 0) + cumulative_sum(append_row(0, log1m_exp(log_z)));
  target += log_x[N];
```

Is this significantly more efficient? If so, it might be preferable, even though IMO it is less readable.

Reply:
Right---if `logistic_lcdf` isn't itself vectorized, then it's tough.

Is that `+= log_x[N]` correct?  If so, it's best to just calculate the single value corresponding to `log_x[N], which I think is just `sum(log1m_exp(log_z))`.

Reply:
Ah---I didn't see `log_x` being used in the target and didn't see that you returned it.  So no, forget my suggestion.